site/src/components/Resources/AgentRow.tsx

```diff
@@ -396,6 +380,51 @@ export const AgentRow: FC<AgentRowProps> = ({
   );
 };
 
+const useAgentLogs = (
+  agentId: string,
+  { enabled, initialData }: { enabled: boolean; initialData?: LineWithID[] },
+) => {
+  const [logs, setLogs] = useState<LineWithID[] | undefined>(initialData);
+  const socket = useRef<WebSocket | null>(null);
+
+  useEffect(() => {
+    if (!enabled) {
+      socket.current?.close();
+      return;
+    }
+
+    socket.current = API.watchWorkspaceAgentLogs(agentId, {
+      // Get all logs
+      after: 0,
```

Comment:
We should not be using `after: 0` all the time.

We intentionally have an endpoint to fetch all logs, and then we should stream afterward, just like before. This hopefully would reduce lagging/flickering on the UI as logs stream in from the WS, because they'd be delivered in a payload first.

Reply:
Hm.. I found having this way faster and better since we don't have to do two operations. It is also easy to understand. In this case, there is no lagging or flickering since when using after 0, all the "past" logs are being sent at once and not one by one.